app/controllers/api/v1/workflows_controller.rb

```diff
@@ -79,15 +82,23 @@ def new_items(resource, relation, value)
     end
   end
 
-  def construct_new_items(item_scope, workflow_project_id)
+  def construct_new_items(item_scope)
     Array.wrap(item_scope).map do |item|
       if item.is_a?(SubjectSet) && !item.belongs_to_project?(workflow_project_id)
-        item.dup.tap do |subject_set|
-          subject_set.project_id = workflow_project_id
-        end
+        duplicate_subject_set_and_subjects(item)
       else
         item
       end
     end
   end
+
+  def duplicate_subject_set_and_subjects(orig_subject_set)
+    orig_subject_set.dup.tap do |subject_set|
+      subject_set.project_id = workflow_project_id
+      subject_set.set_member_subjects_count = 0
+      subject_set.set_member_subjects = orig_subject_set.set_member_subjects.map do |sms|
```

Comment:
I'm not sure if ActiveRecord uses find_each internally for methods like map, but I don't think it does, so this will load all the SetMemberSubjects right away. It might be better to use find_each or find_in_batches to set this up.